### PR TITLE
Teacher lists student emails

### PR DESCRIPTION
**Pull request by FeisEater**

### CODE REVIEW

File: app/controllers/courses_controller.rb [hunk 1]
```diff
@@ -152,6 +152,22 @@ def save_unlocks
     redirect_to manage_unlocks_organization_course_path(@organization, @course), alert: e.to_s
   end
 
+  def student_emails
```
Comment (mpartel) [inline app/controllers/courses_controller.rb]: This functionality seems to overlap with `EmailsController` and `ParticipantsController`. I think we should avoid duplicating the functionality and refactor the former to be suitable for what you need instead.

File: app/controllers/courses_controller.rb [hunk 2]
```diff
@@ -152,6 +152,22 @@ def save_unlocks
...
+    authorize! :teach, @organization
+    add_course_breadcrumb
+    add_breadcrumb('Students', students_organization_course_path(@organization, @course))
+    @students = []
+    @course.submissions.each do |s|
```
Comment (mpartel) [inline app/controllers/courses_controller.rb]: This loop will be quite slow. Maybe use `User.course_students(@course)`.

File: app/views/courses/student_emails.html.erb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      Email
+    </th>
+  </tr>
+  <% @students.each do |student| %>
+    <tr>
+      <td>
+        <%= student.username %>
+      </td>
+      <td>
+        <a href="mailto:<%= student.email %>"><%= student.email %></a>
```
Comment (jamox) [inline app/views/courses/student_emails.html.erb]: Use rather: http://apidock.com/rails/ActionView/Helpers/UrlHelper/mail_to

File: app/controllers/courses_controller.rb [hunk 4]
```diff
@@ -152,6 +152,22 @@ def save_unlocks
...
+      @students << s.user unless @students.include? s.user
+    end
+
+    respond_to do |format|
+      format.html
+      format.text { render text: @students.map { |s| "#{s.email}" }.join("\n") }
+      format.csv { render text: "Username,Email\n" + @students.map { |s| "#{s.username},#{s.email}"}.join("\n") }
+    end
+  end
+
```
Comment (jamox) [inline app/controllers/courses_controller.rb]: Couldn't we easily do this with participants list -- have it courses/:id/participants, with all participants for this course

File: config/routes.rb [hunk 5]
```diff
@@ -25,6 +25,7 @@
         post 'disable'
         get 'manage_unlocks'
         post 'save_unlocks'
+        get 'students', to: 'courses#student_emails'
```
Comment (jamox) [inline config/routes.rb]: could have the participants nested here... see comment above

**Comment by jruo** [comment]
Updated. The ParticipantsController seemed a bit too robust for a simple email list, so I used EmailsController instead.

**Pull request merged**